prismicio/slice-machine
# feat: support nuxt 4

**Resolves**: /

### Description


This PR adds support for the upcoming Nuxt 4 version, it:
- Updates the Nuxt adapter to account for Nuxt 4 changes
- Updates `@slicemachine/init` to detect Nuxt 4 correctly

The main change of Nuxt 4 impacting us is that frontend-related directories will now be nested under an `app` directory (`./pages` is now `./app/pages`), so we have to account for that.

A detailed list of breaking changes and migration strategy is available here: https://nuxt.com/docs/getting-started/upgrade#migrating-to-nuxt-4



### Checklist





### Preview


/

### How to QA [^1]

Nuxt 4 is not released yet, for now you can perform the following:
1. Create a Nuxt 3 app with v4 compat
```
npx nuxi@latest init -t v4-compat
```
2. Run the init command
```
npx @slicemachine/init@alpha.lh-support-nuxt-4
```
3. Check that:
   - `slice-simulator.vue` is created under `app/pages`
   - `app/app.vue` is moved to `app/pages/app.vue`
   - Slice library is pointing to `./app/slices`   






[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.
